packages/astro/src/types/public/integrations.ts

```diff
@@ -59,11 +60,19 @@ export interface AstroRenderer {
 	serverEntrypoint: string;
 }
 
-export type AdapterSupportsKind = 'unsupported' | 'stable' | 'experimental' | 'deprecated';
+export type AdapterSupportsKind =
+	(typeof AdapterFeatureStability)[keyof typeof AdapterFeatureStability];
```

Comment:
Moved the source of truth from the type to the runtime check here, that way they're always accurate